R/paginate_listing.R

```diff
@@ -56,6 +47,24 @@ paginate_listing <- function(lsting,
                              tf_wrap = !is.null(max_width),
                              max_width = NULL,
                              verbose = FALSE) {
+  # process lists of listings
+  if (is(lsting, "list")) {
+    checkmate::assert_true(all(sapply(lsting, is, "listing_df")))
+    cur_call <- match.call(expand.dots = FALSE)
+    if (!"colwidths" %in% names(match.call())) {
+      all_colwidths <- lapply(lsting, propose_column_widths)
+      max_w <- which.max(lapply(all_colwidths, sum))
+      cur_call[["colwidths"]] <- all_colwidths[[max_w]]
+    }
+    lsting_list <- lapply(lsting, function(x_i) {
+      cur_call[["lsting"]] <- x_i
+      eval(cur_call)
+    })
+
+    ret <- unlist(lsting_list, recursive = FALSE)
+    return(ret)
+  }
```

Comment:
I would not add it here. It is already taken care into `paginate_to_mpfs`. I think as_listings (and/or dedicated function) should only make it a list. Then computation of such is devised to exports/paginate_to_mpfs or toString (which can handle already lists as expected)